datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -860,164 +752,85 @@ def remove_complex_dimension_link(  # pylint: disable=too-many-locals
-            },
-            user=current_user.username if current_user else None,
-        ),
-    )
-    session.commit()
-    session.refresh(node)
-    return JSONResponse(
-        status_code=201,
-        content={
-            "message": (
-                f"Dimension link {dimension_node.name} "
-                + (f"(role {link_identifier.role}) " if link_identifier.role else "")
-                + f"to node {node_name} has been removed."
-            )
-            if removed
-            else (
-                f"Dimension link {dimension_node.name} "
-                + (f"(role {link_identifier.role}) " if link_identifier.role else "")
-                + f"to node {node_name} does not exist!"
-            ),
-        },
-    )
+    return remove_dimension_link(session, node_name, link_identifier, current_user)
 
 
 @router.delete("/nodes/{name}/columns/{column}/", status_code=201)
 def delete_dimension_link(
     name: str,
-    column: str,
+    column: str,  # pylint: disable=unused-argument
     dimension: str,
-    dimension_column: Optional[str] = None,
+    dimension_column: Optional[str] = None,  # pylint: disable=unused-argument
     session: Session = Depends(get_session),
     current_user: Optional[User] = Depends(get_current_user),
 ) -> JSONResponse:
     """
     Remove the link between a node column and a dimension node
     """
-    node = get_node_by_name(session=session, name=name)
-    target_column = get_column(node.current, column)
-    if (not target_column.dimension or target_column.dimension.name != dimension) and (
-        not target_column.dimension_column
-        or target_column.dimension_column != dimension_column
-    ):
-        return JSONResponse(
-            status_code=304,
-            content={
-                "message": (
-                    f"No change was made to {column} on node {name} as the "
-                    f"specified dimension link to {dimension} on "
-                    f"{dimension_column} was not found."
-                ),
-            },
-        )
-
-    # Find cubes that are affected by this dimension link removal and update their statuses
-    affected_cubes = get_nodes_with_dimension(
+    return remove_dimension_link(
         session,
-        target_column.dimension,
-        [NodeType.CUBE],
+        name,
+        LinkDimensionIdentifier(dimension_node=dimension, role=None),
+        current_user,
     )
-    if affected_cubes:
-        for cube in affected_cubes:  # pragma: no cover
-            if cube.status != NodeStatus.INVALID:
-                cube.status = NodeStatus.INVALID
-                session.add(cube)
-                session.add(
-                    status_change_history(
-                        node,
-                        NodeStatus.VALID,
-                        NodeStatus.INVALID,
-                        current_user=current_user,
+
+
+@router.post("/nodes/{name}/migrate", status_code=201)  # pragma: no cover
```

Comment:
I know this is temporary, but it may feel safer to call if this endpoint is named clearer, e.g. `migrate_dim_link`.

Reply:
good point, renaming...